pkg/clients/nomad.go

```diff
@@ -248,27 +250,126 @@ func (n *NomadImpl) ParseJob(file string) ([]byte, error) {
 	return jsonJob, nil
 }
 
-// JobStatus returns a string status for the given job
-func (n *NomadImpl) JobStatus(job string) (string, error) {
+// JobRunning returns true when all allocations for a job are running
+func (n *NomadImpl) JobRunning(job string) (bool, error) {
+	jobDetail, err := n.getJobAllocations(job)
+	if err != nil {
+		return false, err
+	}
+
+	if len(jobDetail) < 1 {
+		return false, nil
+	}
+
+	for _, v := range jobDetail {
+		if v["ClientStatus"].(string) != "running" {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+// Endpoints returns a list of endpoints for a cluster
+func (n *NomadImpl) Endpoints(job, group, task string) ([]map[string]string, error) {
+	jobs, err := n.getJobAllocations(job)
+	if err != nil {
+		return nil, err
+	}
+
+	endpoints := []map[string]string{}
+
+	// get the allocation details for each endpoint
+	for _, j := range jobs {
+		r, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/v1/allocation/%s", n.c.APIAddress(), j["ID"]), nil)
+		if err != nil {
+			return nil, xerrors.Errorf("Unable to create http request: %w", err)
+		}
+
+		resp, err := n.httpClient.Do(r)
+		if err != nil {
+			return nil, xerrors.Errorf("Unable to get allocation: %w", err)
+		}
+
+		if resp.Body == nil {
+			return nil, xerrors.Errorf("No body returned from Nomad API")
+		}
+
+		defer resp.Body.Close()
+
+		allocDetail := allocation{}
+		err = json.NewDecoder(resp.Body).Decode(&allocDetail)
+		if err != nil {
+			return nil, err
+		}
+
+		ports := []string{}
+
+		//find the ports used by the task
+		for _, tg := range allocDetail.Job.TaskGroups {
+			if tg.Name == group {
+				for _, t := range tg.Tasks {
+					if t.Name == task {
+						ports = t.Config.Ports
+					}
+				}
+			}
+		}
+
+		ep := map[string]string{}
+		epc := 0
+		for _, p := range ports {
+			// lookup the resources for the ports
+			for _, n := range allocDetail.Resources.Networks {
+				for _, dp := range n.DynamicPorts {
+					if dp.Label == p {
```

Comment:
if you don't set a name in the nomad job for the port, will the label be the string of the port number?

Reply:
I think you always have to set a name as it is a requirement on the `port` stanza